.github/workflows/lint.yml

```diff
@@ -0,0 +1,65 @@
+name: CI
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+  merge_group: # GitHub Merge Queue
+permissions:
+  contents: read
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          persist-credentials: false
+      - uses: julia-actions/setup-julia@5c9647d97b78a5debe5164e9eec09d653d29bd71 # v2.6.1
+        with:
+          version: '1'
+      - name: Run the linter on the users.txt file
+        run: |
+          filename = "users.txt"
+          original_file_contents = read(filename, String)
```

Comment:
This is equivalent, except for the error messages:
```julia
occursin(r"^(@[a-zA-Z0-9\-]*\s*(#.*)?\n)*$", original_file_contents)
```

And this should be equivalent and have even nicer errors (source line annotations)

```julia
fail = false
for (i,line) in enumerate(eachline(filename, keep=true))
    if !occursin(r"^(@[a-zA-Z0-9]+)?(\s*#.*)?\n$", line)
        println("::error file=$filename,line=$i::Line does not match linter regex")
        fail = true
    end
end
if fail
    error("One or more lines failed the linter regex")
end
```